packages/react/src/popover/root/usePopoverRoot.ts

```diff
@@ -75,15 +75,20 @@ export function usePopoverRoot(params: usePopoverRoot.Parameters): usePopoverRoo
     },
   );
 
+  const handleUnmount = useEventCallback(() => {
+    setMounted(false);
+    setOpenReason(null);
+  });
+
   useAfterExitAnimation({
+    enabled: !params.unmountRef,
     open,
     animatedElementRef: popupRef,
-    onFinished: () => {
-      setMounted(false);
-      setOpenReason(null);
-    },
+    onFinished: handleUnmount,
   });
 
+  React.useImperativeHandle(params.unmountRef, () => ({ unmount: handleUnmount }), [handleUnmount]);
```

Comment:
An API with a bunch of imperative methods could be useful in general, not just for this, so it could be named something more generic to future-proof it. This concept may be similar to using hooks, or component stores as in Ariakit. One difference is you can't "read" values in render, it's just for accessing internal methods/state inside event handlers & effects (since it's a ref).